File: Minimum Deletions to Make String K-Special/Minimum Deletions to Make String K-Special.java
// Solution for Minimum Deletions to Make String K-Special in JAVA

class Solution {
    public int minimumDeletions(String word, int k) {
        int[] freq = new int[26];

        for (char ch : word.toCharArray()) {
            freq[ch - 'a']++;
        }

        Arrays.sort(freq);
        int minDeletions = Integer.MAX_VALUE;
        int cumulativeDeletions = 0;

        for (int i = 0; i < 26; i++) {
            int deletionsNeeded = cumulativeDeletions;

            for (int j = 25; j > i; j--) {
                if (freq[j] - freq[i] <= k) break;
                deletionsNeeded += freq[j] - freq[i] - k;
            }

            minDeletions = Math.min(minDeletions, deletionsNeeded);
            cumulativeDeletions += freq[i];
        }

        return minDeletions;
    }
}

File: Minimum Deletions to Make String K-Special/Minimum Deletions to Make String K-Special.py
# Solution for Minimum Deletions to Make String K-Special in PY

class Solution:
    def minimumDeletions(self, word: str, k: int) -> int:
        freq = [0] * 26

        for ch in word:
            freq[ord(ch) - ord('a')] += 1

        freq.sort()
        min_deletions = float('inf')
        cumulative_deletions = 0

        for i in range(26):
            deletions_needed = cumulative_deletions

            for j in range(25, i, -1):
                if freq[j] - freq[i] <= k:
                    break
                deletions_needed += freq[j] - freq[i] - k

            min_deletions = min(min_deletions, deletions_needed)
            cumulative_deletions += freq[i]

        return min_deletions
